pkg/broker/aws_sdk.go

```diff
@@ -72,7 +72,14 @@ func assumeTargetRole(sess *session.Session, params map[string]string, region st
 		return sess, nil
 	}
 
-	targetAccountRoleArn := generateRoleArn(params, accountId)
+	// retrieve AWS partition from instance metadata service
+	partition, err := ec2metadata.New(sess).GetMetadata("/services/partition")
```

Comment:
In some cases the broker may not be running on EC2, or be running in a container that has acceddd to the metadata service blocked. Ideally the partition should be configurable via cli argument, which could override the metadata call. 

Or, at minimum, if ec2 metadata call fails set it to the default `aws` partition

Reply:
I will default to `aws` partition